src/screens/ContactDetail/ContactDetail.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { createRef, useEffect, useState } from 'react';
+import {
+  SafeAreaView,
+  View,
+  ScrollView,
+  KeyboardAvoidingView,
+} from 'react-native';
+
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { CompositeScreenProps } from '@react-navigation/native';
+import { useHeaderHeight } from '@react-navigation/elements';
+import Toast from 'react-native-toast-message';
+import { FormikProps } from 'formik';
+
+import { ProfileStackParams, RootStackParams } from '../../Navigator';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { Contact } from '../../store/types';
+import {
+  getAllContacts,
+  getContactById,
+  removeContact,
+  updateContact,
+} from '../../store/thunks/contacts';
+
+import { selectedContactSelector } from '../../store/selectors/contacts';
+import { ContactHeader } from './components/ContactHeader';
+import { ContactForm } from '../../components/ContactForm';
+import { Button } from '../../components/Button';
+
+import { isIOS } from '../../util/platform';
+import styles from './styles';
+
+export type ContactDetailProps = CompositeScreenProps<
+  NativeStackScreenProps<ProfileStackParams, 'contactDetail'>,
+  NativeStackScreenProps<RootStackParams>
+>;
+
+export const ContactDetail = ({ navigation, route }: ContactDetailProps) => {
+  const dispatch = useAppDispatch();
+  const headerHeight = useHeaderHeight();
+  const { contactId, editContent } = route?.params;
+
+  const formRef = createRef<FormikProps<Contact>>();
+  const [isEditing, setIsEditing] = useState<boolean>(!!editContent);
+
+  const contactData: Contact | undefined = useAppSelector(
+    selectedContactSelector,
+  );
+
+  useEffect(() => {
+    dispatch(getContactById({ id: contactId }));
+  }, []);
+
+  const showToast = (type, errorMessage) =>
+    Toast.show({
+      type: type,
+      text1: errorMessage,
+    });
+
+  const toggleEditStatus = () => setIsEditing(prev => !prev);
+
+  const handleSaveAction = async values => {
+    try {
+      const payload = {
+        id: contactData?.contactId,
+        ...values,
+      };
+      await dispatch(updateContact(payload)).unwrap();
+      dispatch(getAllContacts());
+      showToast('success', 'Contact edited with successful.');
+      toggleEditStatus();
+    } catch (e) {
+      const errorMessage =
+        e?.message ||
+        "The contact wasn't edited successfully, please try again.";
```

Comment:
Message text should be changed, I suggest "Failed to edit contact, please try again"